File: 0347-top-k-frequent-elements/0347-top-k-frequent-elements.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        dic = {}
        for i in nums:
            if i in dic.keys():
                dic[i] += 1
            else:
                dic[i] = 1
        res = dict(sorted(dic.items(), key = itemgetter(1), reverse = True)[:k])
        ans = []
        for i in res:
            ans.append(i)
        return ans

File: 0347-top-k-frequent-elements/0347-top-k-frequent-elements.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        unordered_map<int,int> map;
        for(int num : nums){
            map[num]++;
        }
        vector<int> ans;
        priority_queue<pair<int,int>> pq; 
        for(auto it = map.begin(); it != map.end(); ++it){
            pq.push({it->second, it->first});
            if(pq.size() > map.size() - k){
                ans.push_back(pq.top().second);
                pq.pop();
            }
        } 
        return ans;
    }
};